R/data.table.R

```diff
@@ -557,7 +557,7 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
             if (verbose) cat("on= matches existing key, using key\n")
           } else {
             if (isTRUE(getOption("datatable.use.index"))) {
-              idxName = paste0("__", names(on), sep="", collapse="")  # TODO: wrong, no sep!
```

Comment:
this looks wrong to me anyway... isn't the format `__key1__key2__key3`? so `paste(c('', names(on)), collapse = '__')`?

Reply:
Yes good thought, that is the format with leading `__`, but that is what that line does. The `sep=""` is superfluous given it's `paste0`, but otherwise it looks correct.  `paste0` appends its first argument to all items in its second argument, then `collapse=""` puts them all into one string.
```R
> paste0("__", c("foo","bar"), collapse="")
[1] "__foo__bar"
```